src/main/java/org/openrewrite/java/logging/slf4j/JulParameterizedArguments.java

```diff
@@ -88,6 +88,57 @@ private static J.Literal buildStringLiteral(String string) {
             return new J.Literal(randomId(), Space.EMPTY, Markers.EMPTY, string, String.format("\"%s\"", string), null, JavaType.Primitive.String);
         }
 
+        /**
+         * Create a list of expression representing each element of an array
+         *
+         * @param arrayExpression Either a J.NewArray or a J.Identifier of an array
+         * @param indiceCount     size of the array
+         * @return A List of expression representing each item of the array. If the passed arrayExpression isn't of type J.NewArray or J.Identifier, the arrayExpression is returned alone in a list.
+         */
+        private static List<Expression> splitArrayToExpressions(Expression arrayExpression, int indiceCount) {
+            if (arrayExpression instanceof J.NewArray) {
+                final List<Expression> initializer = ((J.NewArray) arrayExpression).getInitializer();
+                if (initializer == null || initializer.isEmpty()) {
+                    return Collections.emptyList();
+                }
+                return initializer;
+            }
+            if (arrayExpression instanceof J.Identifier && arrayExpression.getType() instanceof JavaType.Array) {
+                List<Expression> arrayAccessExpr = new ArrayList<>(indiceCount);
+                for (int i = 0; i < indiceCount; i++) {
+                    arrayAccessExpr.add(
+                            new J.ArrayAccess(randomId(), Space.EMPTY, Markers.EMPTY, arrayExpression.withPrefix(Space.EMPTY),
+                                    new J.ArrayDimension(randomId(), Space.EMPTY, Markers.EMPTY,
+                                            new JRightPadded<>(
+                                                    new J.Literal(
+                                                            randomId(), Space.EMPTY, Markers.EMPTY, i, String.valueOf(i), null, JavaType.Primitive.Int
+                                                    ), Space.EMPTY, Markers.EMPTY
+                                            )
+                                    ), arrayExpression.getType()
+                            )
+                    );
```

Comment:
```suggestion
                    J.Literal literal = new J.Literal(randomId(), EMPTY, Markers.EMPTY, i, String.valueOf(i), null, JavaType.Primitive.Int);
                    J.ArrayDimension dimension = new J.ArrayDimension(randomId(), EMPTY, Markers.EMPTY, new JRightPadded<>(literal, EMPTY, Markers.EMPTY));
                    J.ArrayAccess arrayAccess = new J.ArrayAccess(randomId(), EMPTY, Markers.EMPTY, arrayExpression.withPrefix(EMPTY), dimension, arrayExpression.getType());
                    arrayAccessExpr.add(arrayAccess);
```

I think by using variables your code is easier to read.